### PR TITLE
Allow saving empty solution to prevent inconsistencies

### PR DESCRIPTION
**Pull request by julian-berbel**

### CODE REVIEW

**Review by luchotc — APPROVED** [review]

File: app/models/assignment.rb [hunk 1]
```diff
@@ -118,7 +118,7 @@ def self.evaluate_manually!(teacher_evaluation)
   end
 
   def content=(content)
-    if content.present?
+    unless exercise.reading?
       self.solution = exercise.single_choice? ? exercise.choice_index_for(content) : content
```
Comment (flbulgarelli) [inline app/models/assignment.rb:122]: That is a huge change! Can't this have subtle unexpected effects? Is this code  fixing an issue?

Comment (julian-berbel) [reply app/models/assignment.rb:122]: it's not really so much fixing an issue as making one more transparent, and also lessening its effects.
Right now, if a user has a passed assignment on an exercise and sends an empty solution, the tests are run and the results are saved, while the solution's content is not, leading to inconsistencies.

Comment (luchotc) [reply app/models/assignment.rb:122]: We could make this PR more clear by adding a test prior to this change in which we show that behavior. What do you think @julian-berbel?

Comment (julian-berbel) [reply app/models/assignment.rb:122]: sounds good!

**Comment by luchotc** [comment]
For me it's ready but i want to hear @flbulgarelli further thoughts

**Comment by luchotc** [comment]
@flbulgarelli has orally approved this one :smile: 
We can rebase it and merge it @julian-berbel

**Comment by flbulgarelli** [comment]
> @flbulgarelli has orally approved this one smile

To sum up, the original code was so wrong that I did don't understand how I could think it was right :stuck_out_tongue:

**Pull request merged**